lib/connextclient/ConnextClient.ts

```diff
@@ -399,10 +400,26 @@ class ConnextClient extends SwapClient {
       assetId,
       preImage: `0x${rPreimage}`,
     });
+    this.expectedIncomingTransfers.delete(rHash);
   }
 
   public removeInvoice = async (rHash: string) => {
-    this.expectedIncomingTransfers.delete(rHash);
+    const expectedIncomingTransfer = this.expectedIncomingTransfers.get(rHash);
+    if (expectedIncomingTransfer) {
+      const { paymentId } = expectedIncomingTransfer;
+      if (paymentId) {
+        // resolve a hashlock with a paymentId but no preimage to cancel it
+        await this.sendRequest('/hashlock-resolve', 'POST', {
+          paymentId,
+          assetId: expectedIncomingTransfer.tokenAddress,
+        });
```

Comment:
1 minor thing - can we add logging here so that it's analogical to `LndClient.removeInvoice`:
```
this.logger.debug(`canceled invoice for ${rHash}`);
```